include/attribute.h

```diff
@@ -33,241 +31,245 @@ struct BagWeapon
-    u8 entrustFail3;
-    u8 shifusWeapon;
-    u8 mightBook1;
-    u8 mightBook2;
-    u8 mightBook3;
-    u8 mightBook4;
-    u8 mightBook5;
-    u8 craftMightBook;
-    u8 mightBook6; // looks wierd. maybe need a fix
-    u8 auditionMachine;
-    u8 waterDragonScale;
-    u8 inviteToTheGreatWar;
-    u8 watchingWarInvite;
-    u8 ex1Invite;
-    u8 ex2Invite;
-    u8 jvhuaChaInStory;
-    u8 luggageToSend;
-    u8 goldFaction99Entrust;
-    u8 zaksHammer;
-    u8 jetEntrustFail;
-    u8 zaksEntrustFail; // where's edi?
-    u8 filler53D;
-    u8 filler53E;
-    u8 filler53F;
-    u8 filler540[0x24];
-    u8 weaponDex[0x1E]; // This is a bit field. 1 = dex lighted, 0 = not lighted. 
-    u8 filler582; // wierd
-    u8 weaponSkillDex1[6];
-    u8 weaponSkillDex2[6]; // maybe a useless checksum, need more evidence
-    u8 weaponSpecialEffectDex1[5];
-    u8 filler594;
-    u8 weaponSpecialEffectDex2[5];
-    u8 filler59A;
-    u8 summonGuardianDex[17];
-    u32 money; // max = 9999999, or 0x0098967F in hex
-    u8 filler5B0[0x14]; // something related to the map. 
-    u32 filler5C4; // something related to the probability of collision with an enemy. 
-    u8 filler5C8[0xE];
-    u16 fishingPoint;
+    /*0x000*/ u8 playerGender; // 00 = male
+    /*0x001*/ u8 playerLevel; 
+    /*0x002*/ u8 forgeLevelList[6]; // blade, axe, spear, boxglove, drill, bow. 00-09
+    /*0x008*/ u16 playerName[0x7];
+    /*0x016*/ u16 unk16[0x7];
+    /*0x024*/ u16 playerMaxHp; // max = 0x270f, but more is ok
+    /*0x026*/ u16 playerCurrentHp;
+    /*0x028*/ u16 playerAtkList[4]; //Atk when equip weapon1, weapon2, weapon3, noweapon, max = 03e7
+    /*0x030*/ u16 filler30; // always 03e7 in memory, wierd
+    /*0x032*/ u16 playerDefList[4];
+    /*0x03A*/ u16 filler3A;
+    /*0x03C*/ u16 playerAglList[4];
+    /*0x044*/ u16 filler44;
+    /*0x046*/ u8 equipWeaponTypeList[3]; // index of equip weapon1, weapon2, weapon3 max = 0x1D, as one can take up to 30 wepons in the bag
+    /*0x049*/ u8 equipItemType;
+    /*0x04A*/ u16 unk4A[6];
+    /*0x056*/ u16 bonusStats; // synced with 020036BA, void?
+    /*0x058*/ u8 unk58;
+    /*0x059*/ u8 unk59;
+    /*0x05A*/ u16 filler5A;
+    /*0x05C*/ u32 playerExp; // max = 00BD95FC
+    /*0x060*/ u32 unk60;
+    /*0x064*/ u8 guardimalType; // 00-04
+    /*0x065*/ u8 guardimalLevel;
+    /*0x066*/ u8 filler66[0xE]; // Something related to guardimal name
+    /*0x074*/ u16 guardimalHp;
+    /*0x076*/ u16 filler76;
+    /*0x078*/ u16 guardimalMag;
+    /*0x07A*/ u16 guardimalAtk;
+    /*0x07C*/ u16 guardimalDef;
+    /*0x07E*/ u16 guardimalAgl;
+    /*0x080*/ u32 filler80;
+    /*0x084*/ u32 guardimalExp;
+    /*0x088*/ u32 guardimalNextExp;
+    /*0x08C*/ u8 battleItemOne;
+    /*0x08D*/ u8 battleItemTwo;
+    /*0x08E*/ u8 battleItemThree;
+    /*0x08F*/ u8 battleItemFour;
+    /*0x090*/ u8 filler90[0x54];
+    /*0x0E4*/ struct BagWeapon bagWeapon[0x1E]; 
+    /*0x42C*/ u8 filler42C[0x1E];
+    /*0x*/ u8 numWoundBand;
```

Comment:
Missing offset of field from here. Bug from your tool?

Reply:
No. Actually I do this manually. I am not sure if I write the struct right, so I think it is a double check to list offsets by hand. (The main reason is I am not familiar with programming of course.T_T)